extensions/guacamole-auth-radius/src/main/java/org/apache/guacamole/auth/radius/RadiusAuthenticationProviderModule.java

```diff
@@ -71,11 +74,28 @@ public RadiusAuthenticationProviderModule(AuthenticationProvider authProvider)
                     || innerProtocol == RadiusAuthenticationProtocol.MSCHAPv1 
                     || innerProtocol == RadiusAuthenticationProtocol.MSCHAPv2) {
             
-            Security.addProvider(new Provider("MD4", 0.00, "MD4 for MSCHAPv1/2 Support") {
-                {
-                    this.put("MessageDigest.MD4", org.bouncycastle.jce.provider.JDKMessageDigest.MD4.class.getName());
-                }
-            });
+            try {
+                Provider md4Provider;
+                Constructor providerConstructor = Provider.class
+                        .getConstructor(String.class, String.class, String.class);
+                if (providerConstructor != null)
+                    md4Provider = (Provider) providerConstructor
+                            .newInstance("MD4", "0.00", "MD4 for MSCHAPv1/2 Support");
```

Comment:
I haven't checked myself, but is "0.00" the correct format for the version string expected by the new constructor?

Reply:
I'm not sure - details on that seem to be few and far between.  But maybe there's an easier way than this, even...
```
try {
    MessageDigest.getInstance("MD4");
}
catch (NoSuchAlgorithmException e)
{
    Security.addProvider(new BouncyCastleProvider());
}
```

Reply:
If the above works, that looks so much better.

Reply:
Yeah...problem is, I don't have much of a way to test it :-).  I can reach out to the folks who reported the issue and see if they're able to test it.

Reply:
Well, I implemented the above code, and verified that the module loads successfully when configured for `mschapv1` or `mschapv2`, which should indicate that it's able to load the MD4 code.  I'm reasonably certain that MD4 support was removed in Java 1.8, which is what I'm running, so I'm pretty sure this is working as intended.  Just hard to say for sure.